contrib/pmsnare/pmsnare.c

```diff
@@ -81,94 +84,93 @@ CODESTARTisCompatibleWithFeature
-
-	 find out if we have a space separated or tab separated for the first item
-	 if tab separated see if the second word is one of our expected tags
-	  if so replace the tabs with spaces so that hostname and syslog tag are going to be parsed properly
-	   optionally replace the hostname at the beginning of the message with one from later in the message
-	  else, wrong message, abort
-	 else, assume that we have a valid timestamp, move over to the syslog tag
-	  if that is tab separated from the rest of the message and one of our expected tags
-	   if so, replace the tab with a space so that it will be parsed properly
-	    optionally replace the hostname at the beginning of the message withone from later in the message 
-
-	*/
+	/* Check if this message is of the type we handle in this parser.
+	 *
+	 * - Find out if the first separator is a tab or a space.
+	 * - If tab-separated, see if the second word is one of our expected tags.
+	 *   - If so, flag as Snare and replace the first tab with space so that hostname and syslog tag parse properly
+	 *   - Else not a snare message.
+	 * - Else assume valid 3164 timestamp, move over to the syslog tag.
+	 * - See if that is tab-separated from one of our expected tags.
+	 *   - If so, flag as Snare.
+	 * - See if either type flagged as Snare.
+	 *   - If so, replace the tab with a space so that it will be parsed properly.
+	 */
+	 
 	snaremessage=0;
-	lenMsg = pMsg->iLenRawMsg - pMsg->offAfterPRI;
 	/* note: offAfterPRI is already the number of PRI chars (do not add one!) */
-	p2parse = pMsg->pszRawMsg + pMsg->offAfterPRI; /* point to start of text, after PRI */
+	lenMsg = pMsg->iLenRawMsg - pMsg->offAfterPRI;
+	/* point to start of text, after PRI */
+	p2parse = pMsg->pszRawMsg + pMsg->offAfterPRI;
 	dbgprintf("pmsnare: msg to look at: [%d]'%s'\n", lenMsg, p2parse);
 	if((unsigned) lenMsg < 30) {
 		/* too short, can not be "our" message */
-		dbgprintf("msg too short!\n");
+		dbgprintf("Message is too short to be Snare!\n");
 		ABORT_FINALIZE(RS_RET_COULD_NOT_PARSE);
 	}
 
-	while(lenMsg && *p2parse != ' ' && *p2parse != '\t' && *p2parse != '#') {
+	/* Check for non-syslog snare format. */
+	while(lenMsg && *p2parse != ' ' && *p2parse != tabRepresentation[0]) {
+		/* Ignore anything before first space/tab/special. */
 		--lenMsg;
 		++p2parse;
 	}
-	dbgprintf("pmsnare: separator [%d]'%s'  msg after the first separator: [%d]'%s'\n", tablength,
-	TabRepresentation,lenMsg, p2parse);
-	if ((lenMsg > tablength) && (*p2parse == '\t' || strncasecmp((char*) p2parse, TabRepresentation ,
-	tablength-1) == 0)) {
-	//if ((lenMsg > tablength) && (*p2parse == '\t' || *p2parse == '#')) {
-	dbgprintf("pmsnare: tab separated message\n");
-		if(strncasecmp((char*) (p2parse + tablength - 1), "MSWinEventLog", 13) == 0) {
-			snaremessage=13; /* 0 means not a snare message, a number is how long the tag is */
+	if ((lenMsg > tabLength) && (strncasecmp((char *)p2parse, tabRepresentation, tabLength) == 0)) {
```

Comment:
This functionality has been moved to endCnfLoad, since the config has not loaded before that point. I looked into laying out the code to match the chronological order the sections are called in, but it's non-trivial and would require many unnecessary changes to other files.